### PR TITLE
TST Adds __repr__ and str to module info

### PR DESCRIPTION
**Pull request by thomasjpfan**
Follow up to https://github.com/pytorch/pytorch/pull/61935

This PR adds `test_repr` to `test_modules`.

cc @albanD @mruberry @jbschlosser @walterddr

### CODE REVIEW

File: test/test_modules.py [hunk 1]
```diff
@@ -108,6 +108,20 @@ def test_factory_kwargs(self, device, dtype, module_info):
                             buffer.dtype, dtype,
                             f'Buffer {name} is of dtype {buffer.dtype} instead of the expected dtype {dtype}')
 
+    @modules(module_db)
+    def test_repr(self, device, dtype, module_info):
+        # Test module can be represted with repr and str without errors.
```
Comment (jbschlosser) [inline test/test_modules.py:113]: typo nit: represted -> represented

**Review by jbschlosser — COMMENTED** [review]
Thanks for adding this! Test logic checks out, although this does bring up a tradeoff between test time and test granularity. That is, the `__repr__()` / `str()` tests could be merged into another set of tests to avoid the additional overhead of module instantiation. For now, I like this better since it's more modular, and we can look into merging if necessary later on.

**Review by jbschlosser — APPROVED** [review]
LGTM! Mind doing a quick rebase?

File: test/test_modules.py [hunk 2]
```diff
@@ -110,6 +110,20 @@ def test_factory_kwargs(self, device, dtype, module_info):
...
                             f'Buffer {name} is of dtype {buffer.dtype} instead of the expected dtype {dtype}')
 
+    @modules(module_db)
+    def test_repr(self, device, dtype, module_info):
+        # Test module can be represented with repr and str without errors.
+        module_cls = module_info.module_cls
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=False)
+        for module_input in module_inputs:
+            args, kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
```
Comment (mruberry) [inline test/test_modules.py:120]: Drive-by fyi: one thing to be considerate of in the overall design of ModuleInfo testing is test time. I'm not sure how long running through all the modules and their inputs takes but it might makes sense to glom multiple tests together to avoid having to do it so often

File: test/test_modules.py [hunk 3]
```diff
@@ -110,6 +110,20 @@ def test_factory_kwargs(self, device, dtype, module_info):
                             buffer.dtype, dtype,
                             f'Buffer {name} is of dtype {buffer.dtype} instead of the expected dtype {dtype}')
 
+    @modules(module_db)
+    def test_repr(self, device, dtype, module_info):
+        # Test module can be represented with repr and str without errors.
+        module_cls = module_info.module_cls
```
Comment (mruberry) [inline test/test_modules.py:116]: Is this pattern common enough that it's worth abstracting it?
